docs/debugger/dap.md

```diff
@@ -0,0 +1,34 @@
+# Debug Adapter Protocol
+
+[Debug Adapter Protocol](https://www.graalvm.org/latest/tools/dap/) is yet
+another instrument available for a Truffle language. The DAP is a native
+protocol for VSCode and as such, works only via VSCode. To start Enso with DAP
+server waiting for a client to attach, launch enso via:
+
+```
+env JAVA_OPTS='-Dpolyglot.dap' ./built-distribution/enso-engine-*/enso-*/bin/enso --run *.enso
+```
+
+Once DAP server is started and ready for a client to be attached, the following
+output will be printed:
+
+```
+[Graal DAP] Starting server and listening on /127.0.0.1:4711
+```
+
+In VSCode, create a new
+[Launch configuration](https://code.visualstudio.com/docs/editor/debugging#_launch-configurations)
+of type `node`, for example:
+
+```json
+{
+  "name": "Enso debug",
+  "type": "node",
+  "debugServer": 4711,
+  "request": "attach"
+}
+```
+
+And launch it.
```

Comment:
I tried following this tutorial and I got lost at this `And launch it` step.

How do I launch a debug configuration added in the way as explained in this doc? I cannot find any information on how to do that. Could you please clarify that step? Maybe we should have a screenshot/screencast showing how to do it?

Reply:
I've tried pressing 'Add configuration' on this config file, adding some Node entry I ended up with:

![image](https://github.com/enso-org/enso/assets/1436948/2da29c8f-a435-4bbd-ad4a-49b744ecc321)

This finally made a button show up that I was able to run the Attach config. But it failed 
on Enso side with:
```
[Graal DAP] Starting server and listening on localhost/127.0.0.1:4711
[Graal DAP] Client connected on /127.0.0.1:59320
[ERROR] [2023-11-20T18:16:40+01:00] [dap] Error while processing an incomming message: Missing header Content-Length: in input.
```

Reply:
Sorry, I thought this step was clear - I rewrote it in a123cd0d9f4b04d05aae7a5231efba554062188f : Just start debugging and make sure that the correct configuration is used.

Reply:
It is still not clear to me.

With the config you have there, I still have no debug configurations nor play button:
![image](https://github.com/enso-org/enso/assets/1436948/820f1ca1-6fef-4a38-9ac3-7ed7ff8fe12d)

Ah but I see my problem! What you provided is an example configuration to paste there, instead I replaced the whole `launch.json` with the content you provided.

Please clarify it that I need to add it to the list, not replace the file. Or better - add an example of what the file could look like overall.

Once my file was
```json
{
    "configurations": [
        {
            "name": "Enso debug",
            "type": "node",
            "debugServer": 4711,
            "request": "attach"
          }
    ]
}
```
then I was able to happily debug Enso using DAP in VS Code.

Maybe the example we provide should look like this instead of just showing a section of the file?

I assume if the user has other existing configurations - they will know how to merge it. But for users (like me) that have never set-up VSC's `launch.json` config, it is unclear what data to put in there - so having an example of the full file is much easier. A screenshot of what it is supposed to look like could also help:
![image](https://github.com/enso-org/enso/assets/1436948/b939e2a8-7d8e-4286-b7d3-61fbfbd99ce6)